int21.c

```diff
@@ -155,6 +172,29 @@ void int21(uc_engine *uc)
                 break;
             }
 
+        case 0x0a: // buffered keyboard input
+            {
+                uint16_t r_dx, r_ds;
+                uint8_t max_buf = 0;
+                char *buf = NULL;
+                char *str = NULL;   
+
+                uc_reg_read(uc, UC_X86_REG_DX, &r_dx);
+                uc_reg_read(uc, UC_X86_REG_DS, &r_ds);
+                uc_mem_read(uc, r_dx, &max_buf, 1);
```

Comment:
at line 190 you have the address right when writing to memory, but then why at line 184 you dont do the same address calculation for memory reading?